dids/didcore/document.go

```diff
@@ -97,6 +101,87 @@ func (d *Document) AddVerificationMethod(method VerificationMethod, opts ...AddV
 	}
 }
 
+// VMSelector is an interface that can be implemented to provide a means to select
+// a specific verification method from a DID Document.
+type VMSelector interface {
+	selector()
+}
+
+// Purpose can be used to select a verification method with a specific purpose.
+type Purpose string
+
+func (p Purpose) selector() {}
+
+// ID can be used to select a verification method by its ID.
+type ID string
+
+func (i ID) selector() {}
+
+// SelectVerificationMethod takes a selector that can be used to select a specific verification
+// method from the DID Document. If no selector is provided, the first verification method
+// is returned
+//
+// The selector can either be an ID or a Purpose. If a Purpose is provided, the first verification
+// method in the DID Document that has the provided purpose will be used to sign the payload.
+func (d *Document) SelectVerificationMethod(selector VMSelector) (VerificationMethod, error) {
+	if len(d.VerificationMethod) == 0 {
+		return VerificationMethod{}, fmt.Errorf("no verification methods found")
+	}
+
+	if selector == nil {
+		return d.VerificationMethod[0], nil
+	}
+
+	var vmID string
+	switch s := selector.(type) {
+	case Purpose:
+		switch purpose := Purpose(s); purpose {
+		case "assertionMethod":
```

Comment:
These should be typed constants...then users can just pass a constant in to the function.

eg.

```go
const PurposeAssertion Purpose = "assertionMethod"
const PurposeAuthentication Purpose = "authentication"
const PurposeCapabilityDelegation Purpose  = "capabilityDelegation"
// ...

doc.SelectVerificationMethod(PurposeAuthentication)
```

Reply:
done: https://github.com/TBD54566975/web5-go/commit/d5b306b593e026205d4492d93b29f6a780333edb